src/plugin/field.ts

```diff
@@ -184,24 +226,28 @@ export const updateFieldsFromNode = <FDesc extends FieldDescriptions<string>>({
         const fields =
           currentFieldIndex !== -1
             ? field.children[currentFieldIndex]
-            : getFieldsFromNode({
+            : getFieldsFromNode<FDesc, ElementNames, ESpecMap>({
                 node: childNode,
-                fieldDescriptions: field.description.fields,
+                fieldDescriptions: field.description.fields as FDesc,
                 view,
                 getPos,
                 serializer,
                 offset: accumulatedOffset,
                 innerDecos,
+                getElementDataFromNode,
+                transformElementOut,
               });
 
         const newFieldsForChild = updateFieldsFromNode({
           node: childNode,
-          fields,
+          fields: fields as FieldNameToField<FDesc>,
```

Comment:
why is a cast needed here?

Reply:
(We talked through the type casts off Github and decided to work on them in a follow-up PR)